src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -134,27 +108,77 @@ public BufferedRecords(
         return flushed;
     }
 
+    private void prepareStatement() throws SQLException {
+        final String sql = writeSql();
+
+        log.debug("Prepared SQL {} for insert mode {}", sql, config.insertMode);
+
+        close();
+        preparedStatement = connection.prepareStatement(sql);
+        preparedStatementBinder = dbDialect.statementBinder(
+                preparedStatement,
+                config.pkMode,
+                currentSchemaPair,
+                fieldsMetadata,
+                config.insertMode
+        );
+    }
+
+    private String writeSql() {
+        final String sql;
```

Comment:
No need for this variable, the created value can be `return`ed in both if-else branches,

Reply:
Method completely removed.